compiler/optimizations/gpuTransforms.cpp

```diff
@@ -807,10 +865,17 @@ bool GpuKernel::isCallToPrimitiveWeShouldNotCopyIntoKernel(CallExpr *call) {
      call->isPrimitive(PRIM_GPU_SET_BLOCKSIZE));
 }
 
-void GpuKernel::populateBody(CForLoop *loop, FnSymbol *outlinedFunction) {
+void GpuKernel::populateBody(FnSymbol *outlinedFunction) {
   std::set<Symbol*> handledSymbols;
 
-  for_alist(node, loop->body) {
+  // Some of the conditions below are intended to check for "things that
+  // only happen in the loop body". However, the pre-LICM cloning means
+  // there are actually two loop bodies. Use both for these checks to avoid
+  // throwing off various checks.
```

Comment:
What was a case where you needed to worry about `cloneOfLoop`'s body? It looks to me like we shouldn't transform that loop into a kernel at all?

Reply:
More context for my question: one of the ideas that I believe I mentioned way in the past is if we were to do LICM more aggressively for array metadata, we may be able to hoist some pesky stuff that we need to do for distributed arrays. That stuff (that's not hoisted with this PR, I checked, and it is fine as it is more complicated) makes loops GPU-ineligible and it is the primary blocker for distributed array execution.

IOW, if we have two clones of the loop we may make LICM hoist stuff out of the loop that would otherwise throw off GPU transformation.

Reply:
I think the case is that certain loop index variables have definitions outside the loop body, but are only used inside the loop. The code here calls them "degenerate outer references". The logic to determine if something is degenerate is "is this variable only ever assigned and used inside the loop body?", but now there are two loops, so the question should be, "are is this variable only ever assigned in the two loops' bodies".

Reply:
Ugh... I understand.